packages/site/cms-site/src/blocks/YouTubeVideoBlock.tsx

```diff
@@ -40,6 +40,25 @@ export const YouTubeVideoBlock = withPreview(
     }: YouTubeVideoBlockProps) => {
         const [showPreviewImage, setShowPreviewImage] = useState(true);
         const hasPreviewImage = !!(previewImage && previewImage.damFile);
+        const iframeRef = useRef<HTMLIFrameElement | null>(null);
+        const inViewRef = useRef(null);
+        const inView = useIsElementInViewport(inViewRef);
+
+        const pauseYoutubeVideo = () => {
+            if (iframeRef.current?.contentWindow) {
+                iframeRef.current.contentWindow.postMessage(`{"event":"command","func":"pauseVideo","args":""}`, "*");
+            }
+        };
+
+        const playYoutubeVideo = () => {
+            if (iframeRef.current?.contentWindow) {
+                iframeRef.current.contentWindow.postMessage(`{"event":"command","func":"playVideo","args":""}`, "*");
+            }
+        };
+
+        useEffect(() => {
+            inView && autoplay ? playYoutubeVideo() : pauseYoutubeVideo();
+        }, [autoplay, inView]);
```

Comment:
useIsElementInViewport has a local state and re-renders the Video Block once it gets into view. But you don't have to re-render, as you only call the JS api inside the iframe.

You could argue that performance doesn't matter here, it's only re-rendered once - but I'd argue that it makes the code more complicated than it has to be.

Reply:
How would you design useIsElementInViewport to remove the state and effect? Something like this?

```tsx
useIsElementInViewport(ref, (inView) => {
  if(autoplay) {
    if(inView) {
      playYouTubeVideo();
    } else {
      pauseYouTubeVideo();
    }
  }
});
```

Reply:
Is it better like this?
23411e5d2970a9d1be87ed5d170dd5c05db7aecb

Reply:
> Is it better like this?

yes, much simpler - isn't it?